### PR TITLE
Add .gitignore

### PR DESCRIPTION
**Pull request by gsserge**
Signed-off-by: Sergii Glushchenko <gsserge@amazon.com>

### CODE REVIEW

**Review by jiangliu — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/target
+**/*.rs.bk
+Cargo.lock
```
Comment (lauralt) [inline .gitignore:3]: I don't think we want to ignore Cargo.lock changes for all of the rust-vmm repositories, i.e. vmm-reference.

Comment (jiangliu) [reply .gitignore:3]: Most projects of rust-vmm are library crates, so it may be a reasonable default:)

Comment (gsserge) [reply .gitignore:3]: I'm not 100% sure how GitHub templates work, but unless they are being continuously applied to the *existing* crates, then treating a new crate to be a library is reasonable. `vmm-reference` is just an exception, and I would like to make the general case more comfortable.

Comment (lauralt) [reply .gitignore:3]: In this case can we leave a comment here and mention that `Cargo.lock` should be removed from this file for binary crates? If you don't remove it from .gitignore and try to commit the Cargo.lock you will get the following error:
```
The following paths are ignored by one of your .gitignore files:
Cargo.lock
Use -f if you really want to add them.
````
which might be confusing for new contributors.

**Review by andreeaflorescu — APPROVED** [review]

**Pull request merged**